proposals/Proposal: add cookies.removeAll() method.md

```diff
@@ -0,0 +1,157 @@
+>
+>A  `string`  representing the name of the cookie to remove.
+>
+>[`partitionKey`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#partitionkey)  optional:
+>
+>An object representing the storage partition containing the cookie.
+>
+>[`topLevelSite`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#toplevelsite)  optional:
+>
+>A  `string`  representing the first-party URL of the top-level site storage partition containing the cookie.
+>
+>[`storeId`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#storeid)  optional:
+>
+>A  `string`  representing the ID of the cookie store to find the cookie in. If unspecified, the cookie is looked for by default in the current execution context's cookie store.
+>
+>[`url`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/cookies/remove#url) optional:
+>
+>A  `string`  representing the URL associated with the cookie.
+
+##### Return
+
+> `Promise<Cookie[]>`:
+> 
+> All the cookies that were removed. If list is empty then no cookies were removed.
+
+
+### Behavior
+
+The API will remove all cookies that match the `details` object parameter with the exception of the cases outlined in the implementation details.
+If the extension does not have  [host permissions](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/manifest.json/permissions#host_permissions)  for this URL, the API call will fail.
+
+### New Permissions
+No  new  permissions  are  required.
+
+### Manifest File Changes
+No new manifest fields are required.
+
+## Security and Privacy
+Incorrect usage can lead to the removal of cookies that are not intended.
+
+### Exposed Sensitive Data
+None
+
+### Abuse Mitigations
+To remove a cookie, the extension must have host permissions for the `url` associated with the cookie.
+
+
+## Alternatives
+None
+
+### Existing Workarounds
+
+The behavior introduced by this API can be replicated by using a combination of `cookies.getAll()` and `cookies.remove()`.
+
+### Open Web API
+
+The cookies API is specific to extensions
+
+## Implementation Notes
+
+Host permissions are required for this API.
+#### Special cases:
+```cookies.removeAll({})```
+
+>An empty details object will result in an error. 
+
+```
+cookies.removeAll({
+    partitionKey:{}})
+```
+
+>An empty `partitionKey` object as the only value in the details object, will result in all partitioned cookies being removed.
+
+```
+cookies.removeAll({
+    name: “example”,
+    url: “https://example.com”,
+    partitionKey:{}
+})
+```
+
+>An empty `partitionKey` object with other values populated, will result in both unpartitioned and partitioned cookies that also match the other values provided will be removed.
+
+```
+cookies.removeAll({
+    topLevelSite: “https://example.com”
```

Comment:
Why is topLevelSite a separate key? I would expect it to be a member of partitionKey.

Reply:
Having `topLevelSite` be a separate (optional) parameter addresses the second motivation "Removal of cookies associated with a topLevelSite". 

This could be replaced by having the developer pass in a partitionKey that contains the `toplevelSite` and no value for `hasCrossSiteAncestor`, such as`partitionKey:{topLevelSite: "https/example.com"}`.  But I wanted to provide an easier option for developers who may not be as familiar with the intricacies of partitioned cookies.